experimental/SetPartitions/src/LinearSetPartition.jl

```diff
@@ -0,0 +1,235 @@
+"""
+    LinearSetPartition
+
+LinearSetPartition represents a linear combination of set-partitions. See Chapter 5 in [Gro20](@cite).
+"""
+struct LinearSetPartition{S <: AbstractPartition, T <: RingElement}
+    coefficients :: Dict{S, T}
```

Comment:
I think it would be sensible to store the base ring here as well, as it cannot be reconstructed in all cases (e.g. empty linear comb), and provide it using `base_ring`. This also enables checking the base ring in operations (see comments below)